src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -325,10 +373,90 @@ internal static bool ValidateParentTagName(string parentTag, ErrorSink errorSink
                     errorSink: errorSink);
         }
 
-        private static bool ValidateName(
-            string name,
-            bool targetingAttributes,
-            ErrorSink errorSink)
+        private static bool ValidateRequiredAttribute(string attribute, ErrorSink errorSink)
```

Comment:
Still lots of duplication between this method and `GetRequiredAttributeDescriptors()`. Suggest reordering execution to create `TagHelperRequiredAttributeDescriptor`s (slightly more carefully) then remove invalid required attributes. This method would then take a `TagHelperRequiredAttributeDescriptor` on input and primarily check its `Name` and `Selector` values.